### DISCUSSION TITLE
turbotrace: Support path aliases in `tsconfig.json`

### DISCUSSION THREAD

**Discussion by pan93412** [vercel/next.js#44623] [Ideas]
### Describe the feature you'd like to request

Minimum reproducible repository: <https://github.com/pan93412/turbotrace-issue-221214/tree/alias>

Turbotrace should read the project's `tsconfig.json` to find out the aliases.

```plain
error - [resolve]
  pages/index.tsx
    Error resolving esm request
    unable to resolve module "@components/hi"

  ... [1 more paths] are hidden, run with --show-all to show all
``` 

### Describe the solution you'd like

1. `turbotrace` should read the `tsconfig.json` and parse the wildcard under `compilerOptions.paths`.
2. For larger projects or monorepos, we may also need to read the inherited `tsconfig.json` specifying under the `extend` field.
3. Next.js should pass the `tsconfig.json` of this project to `turbotrace`, so the way to discover `tsconfig.json` can be consistent.

### Describe alternatives you've considered

For the point 3, I am not sure if `turbotrace` should read `tsconfig.json` by themselves or receive from Next.js. Considering the `tsconfig.json` may be extended, I think that may be better to read by `turbotrace` itself.

Not really sure how Next.js handles this issue. I guess Next.js uses the [swc's solution](https://github.com/swc-project/swc/pull/1834)?

**Comment by ahmedkamalio** [comment]
Any news on this? I see that the discussion is not very active, will it be considered?
